### PR TITLE
develprocess: describe host-installer handling as a special case

### PR DESCRIPTION
**Pull request by ydirson**

### CODE REVIEW

File: docs/develprocess.md [hunk 1]
```diff
@@ -221,6 +221,72 @@ We chose to base our `qemu-dp` repository on a fork of the upstream `qemu` [repo
...
+organized for best upstreaming them, which means using per-topic
+branches based on an upstream revision: those can then be used for an
+upstream PR, and for merging into an XCP-ng release.  Constructing an
+XCP-ng `host-installer` release then is done by starting from the
+upstream tag used by the matching XenServer release, and merging those
+topic branches.
+
+A `small-patches` branch is used to hold small non-upstreamable
+patches until we have decided what should be proposed upstream to
+replace them.  This one is merged first (as essentially made of older
```
Comment (benjamreis) [inline docs/develprocess.md:237]: We don't use double space.
```suggestion
patches until we have decided what should be proposed upstream to
replace them. This one is merged first (as essentially made of older
```

File: docs/develprocess.md [hunk 2]
```diff
@@ -221,6 +221,72 @@ We chose to base our `qemu-dp` repository on a fork of the upstream `qemu` [repo
   * If we really need to diverge a little from upstream, on a temporary dev branch `dev-NEXTXCPNGVERSION` (e.g. `dev-8.3`), based either on the current maintenance branch, or on a branch that is likely to be the one used in the next release (we don't always know!)
   * Next maintenance branch directly (e.g. `2.12.0-8.3`) once the upstream SRPMs have been released
 
+#### Special case: `host-installer`
+
+`host-installer` has an upstream repo, and we have a large number of
+modifications that have not yet made it upstream.  Changes are
+organized for best upstreaming them, which means using per-topic
+branches based on an upstream revision: those can then be used for an
+upstream PR, and for merging into an XCP-ng release.  Constructing an
```
Comment (benjamreis) [inline docs/develprocess.md:230]:
```suggestion
branches based on an upstream revision: those can then be used for an
upstream PR, and for merging into an XCP-ng release. Constructing an
```

File: docs/develprocess.md [hunk 3]
```diff
@@ -221,6 +221,72 @@ We chose to base our `qemu-dp` repository on a fork of the upstream `qemu` [repo
   * If we really need to diverge a little from upstream, on a temporary dev branch `dev-NEXTXCPNGVERSION` (e.g. `dev-8.3`), based either on the current maintenance branch, or on a branch that is likely to be the one used in the next release (we don't always know!)
   * Next maintenance branch directly (e.g. `2.12.0-8.3`) once the upstream SRPMs have been released
 
+#### Special case: `host-installer`
+
+`host-installer` has an upstream repo, and we have a large number of
+modifications that have not yet made it upstream.  Changes are
```
Comment (benjamreis) [inline docs/develprocess.md:227]:
```suggestion
`host-installer` has an upstream repo, and we have a large number of
modifications that have not yet made it upstream. Changes are
```

File: docs/develprocess.md [hunk 4]
```diff
@@ -221,6 +221,72 @@ We chose to base our `qemu-dp` repository on a fork of the upstream `qemu` [repo
...
+XCP-ng `host-installer` release then is done by starting from the
+upstream tag used by the matching XenServer release, and merging those
+topic branches.
+
+A `small-patches` branch is used to hold small non-upstreamable
+patches until we have decided what should be proposed upstream to
+replace them.  This one is merged first (as essentially made of older
+changes).
+
+##### adding a new feature
```
Comment (benjamreis) [inline docs/develprocess.md:240]:
```suggestion
##### Adding a new feature
```

File: docs/develprocess.md [hunk 5]
```diff
@@ -221,6 +221,72 @@ We chose to base our `qemu-dp` repository on a fork of the upstream `qemu` [repo
...
+dependencies, it will be clearer to `merge --no-ff` each of those
+other topics in our new topic branch.
+
+This new branch will then be used to open a PR, usually against the
+upstream repo.  In the case of an upstream PR progressing too slowly,
+we may decide to open a PR onto an XCP-ng `host-installer` branch so
+it can be included in a release, and it will join the pool of topic
+branches described above.
+
+##### fixing/extending a topic branch
```
Comment (benjamreis) [inline docs/develprocess.md:255]:
```suggestion
##### Fixing/extending a topic branch
```

File: docs/develprocess.md [hunk 6]
```diff
@@ -221,6 +221,72 @@ We chose to base our `qemu-dp` repository on a fork of the upstream `qemu` [repo
...
+patches until we have decided what should be proposed upstream to
+replace them.  This one is merged first (as essentially made of older
+changes).
+
+##### adding a new feature
+
+A new feature gets a new topic branch, usually based on an upstream
+release -- though in some situations some features will depend on
+another topic branch that has not been merged yet.  In this case the
+branch can be based on its dependency branch.  In the case of multiple
```
Comment (benjamreis) [inline docs/develprocess.md:245]:
```suggestion
another topic branch that has not been merged yet. In this case the
branch can be based on its dependency branch. In the case of multiple
```

File: docs/develprocess.md [hunk 7]
```diff
@@ -221,6 +221,72 @@ We chose to base our `qemu-dp` repository on a fork of the upstream `qemu` [repo
...
+dependencies, it will be clearer to `merge --no-ff` each of those
+other topics in our new topic branch.
+
+This new branch will then be used to open a PR, usually against the
+upstream repo.  In the case of an upstream PR progressing too slowly,
```
Comment (benjamreis) [inline docs/develprocess.md:250]:
```suggestion
upstream repo. In the case of an upstream PR progressing too slowly,
```

File: docs/develprocess.md [hunk 8]
```diff
@@ -221,6 +221,72 @@ We chose to base our `qemu-dp` repository on a fork of the upstream `qemu` [repo
...
+branch", which is to say they should be based on the last commit of
+that topic branch that was merged in the current XCP-ng branch.  A PR
+is then openned for merging into this current XCP-ng branch.
+
+When introducing a fix, we should keep in mind whether it is
+meaningful on its own, or is meant to be squashed the next time this
+topic branch is rebased, and use the standard git conventions to mark
+those, e.g. using `git commit --squash`.
+
+##### upgrading to a new upstream
```
Comment (benjamreis) [inline docs/develprocess.md:267]:
```suggestion
##### Upgrading to a new upstream
```

File: docs/develprocess.md [hunk 9]
```diff
@@ -221,6 +221,72 @@ We chose to base our `qemu-dp` repository on a fork of the upstream `qemu` [repo
...
+upstream repo.  In the case of an upstream PR progressing too slowly,
+we may decide to open a PR onto an XCP-ng `host-installer` branch so
+it can be included in a release, and it will join the pool of topic
+branches described above.
+
+##### fixing/extending a topic branch
+
+Fixing of existing topic branches should be done "on the topic
+branch", which is to say they should be based on the last commit of
+that topic branch that was merged in the current XCP-ng branch.  A PR
```
Comment (benjamreis) [inline docs/develprocess.md:259]:
```suggestion
that topic branch that was merged in the current XCP-ng branch. A PR
```

File: docs/develprocess.md [hunk 10]
```diff
@@ -221,6 +221,72 @@ We chose to base our `qemu-dp` repository on a fork of the upstream `qemu` [repo
...
+is then openned for merging into this current XCP-ng branch.
+
+When introducing a fix, we should keep in mind whether it is
+meaningful on its own, or is meant to be squashed the next time this
+topic branch is rebased, and use the standard git conventions to mark
+those, e.g. using `git commit --squash`.
+
+##### upgrading to a new upstream
+
+To stay switch to a new upstream version `vUPSTREAMVERSION`
```
Comment (benjamreis) [inline docs/develprocess.md:269]:
```suggestion
To switch to a new upstream version `vUPSTREAMVERSION`
```

File: docs/develprocess.md [hunk 11]
```diff
@@ -221,6 +221,72 @@ We chose to base our `qemu-dp` repository on a fork of the upstream `qemu` [repo
...
+(`v10.10.5`), we want a new branch `UPSTREAMVERSION-XCPNGVERSION`
+(`10.10.5-8.3`) based on this tag and including a merge of on every
```
Comment (benjamreis) [inline docs/develprocess.md:271]:
```suggestion
(`10.10.5-8.3`) based on this tag and including a merge of every
```

**Review by benjamreis — CHANGES_REQUESTED** [review]
Except minor form details commented - LGTM

File: docs/develprocess.md (hunk 1 above)
Comment (ydirson) [reply docs/develprocess.md:237]: Damned we need a checker for this, I've been hardwired decades ago to use doublespaces...

Comment (benjamreis) [reply docs/develprocess.md]: IIRC we started to work on something like that but it still is a WIP :thinking: 
But yeah a little checker wouldn't hurt!

**Review by benjamreis — APPROVED** [review]

**Review by Wescoeur — APPROVED** [review]

File: docs/develprocess.md [hunk 12]
```diff
@@ -221,6 +221,72 @@ We chose to base our `qemu-dp` repository on a fork of the upstream `qemu` [repo
...
+we may decide to open a PR onto an XCP-ng `host-installer` branch so
+it can be included in a release, and it will join the pool of topic
+branches described above.
+
+##### Fixing/extending a topic branch
+
+Fixing of existing topic branches should be done "on the topic
+branch", which is to say they should be based on the last commit of
+that topic branch that was merged in the current XCP-ng branch. A PR
+is then openned for merging into this current XCP-ng branch.
```
Comment (stormi) [inline docs/develprocess.md:260]:
```suggestion
is then opened for merging into this current XCP-ng branch.
```

File: docs/develprocess.md [hunk 13]
```diff
@@ -221,6 +221,72 @@ We chose to base our `qemu-dp` repository on a fork of the upstream `qemu` [repo
   * If we really need to diverge a little from upstream, on a temporary dev branch `dev-NEXTXCPNGVERSION` (e.g. `dev-8.3`), based either on the current maintenance branch, or on a branch that is likely to be the one used in the next release (we don't always know!)
   * Next maintenance branch directly (e.g. `2.12.0-8.3`) once the upstream SRPMs have been released
 
+#### Special case: `host-installer`
```
[the remark this answers is not in the document]
Comment (ydirson) [reply docs/develprocess.md:224]: In 2 occurrences it is the sole example, any other example to replace it, or just drop?

[the remark this answers is not in the document]
Comment (stormi) [reply docs/develprocess.md:224]: Just drop the example, but leave the rules in place for now.

**Review by stormi — CHANGES_REQUESTED** [review]
Marking as "changes requested" so that we don't merge by mistake.

**Pull request merged**